src/engine/strat_engine/thindev.rs

```diff
@@ -77,4 +79,68 @@ impl ThinDev {
         try!(dm.device_remove(&DevId::Name(&self.name), DmFlags::empty()));
         Ok(())
     }
+
+    pub fn path(&self) -> EngineResult<Option<PathBuf>> {
+        match self.dev_info {
+            Some(ref di) => return Ok(di.device().path().clone()),
+            None => return Err(EngineError::Engine(ErrorEnum::Invalid, "No dev_info".into())),
+        }
+    }
+
+    pub fn create_fs(&mut self) -> EngineResult<()> {
+        let path_result = self.path().unwrap().unwrap();
+        let dev_path = Path::new(&path_result);
+
+        debug!("Create filesystem for : {:?}", dev_path);
+        let output = try!(Command::new("mkfs.xfs")
+            .arg("-f")
+            .arg(dev_path)
+            .output());
+
+        if output.status.success() {
+            debug!("Created xfs filesystem on {:?}", dev_path)
+        } else {
+            let message = String::from_utf8_lossy(&output.stderr);
+            debug!("stderr: {}", message);
+            return Err(EngineError::Engine(ErrorEnum::Error, message.into()));
+        }
+        Ok(())
+    }
+
+    pub fn mount_fs(&mut self, mount_point: &Path) -> EngineResult<()> {
+        let path_result = self.path().unwrap().unwrap();
```

Comment:
Yeah, it was a Option wrapped in a Result.  I updated to simplify and added try!